app/lib/logic/contracting.rb

```diff
@@ -44,8 +44,7 @@ def provider_builds_application_for(buyer, application_plan, application_attrs =
         application.validate_contract_hierarchy!
         application.validate_human_edition!
 
-        application.plan_id = application_plan.id
-        application.attributes = application_attrs
+        application.attributes = application_attrs.merge('plan_id' => application.plan_id)
```

Comment:
```suggestion
        # `#merge` works here because `application_attrs` is a `ActionController::Parameters`; with a normal `Hash` this will break due to old mass-assignement protection
        application.attributes = application_attrs.merge('plan_id' => application.plan_id)
```

Reply:
Another option:

```ruby
  attrs = application_attrs.dup
  application.plan_id = attrs.delete(:plan_id)
  application.attributes = attrs
```

Reply:
In the future, let's try to move this to Rails strong parameters and avoid the headache.